lightning/src/routing/router.rs

```diff
@@ -893,7 +893,7 @@ where L::Target: Logger {
 							}
 
 							let path_penalty_msat = $next_hops_path_penalty_msat
-								.checked_add(scorer.channel_penalty_msat($chan_id.clone(), &$src_node_id, &$dest_node_id))
+								.checked_add(scorer.channel_penalty_msat($chan_id.clone(), amount_to_transfer_over_msat, *available_liquidity_msat, &$src_node_id, &$dest_node_id))
```

Comment:
I think the channel capacity is supposed to be in `satoshis` not `msat`? Similar below

Reply:
I updated this and also changed the last-hop behavior to be completely different - `u64::max_value()`, cause, like, its really "unlimited value" as far as we're concerned.